src/services/fileStorage/proxy-service.js

```diff
@@ -142,6 +143,23 @@ const getRefOwnerModel = async (owner) => {
 	return refOwnerModel;
 };
 
+/**
+ * returns the schoolId of the given file owner
+ * @param {('user'|'course'|'teams'))} ownerType type of the file owner
+ * @param {ObjectId} ownerId id of the owner
+ * @returns {Promise<ObjectId>} schoolId of the owner
+ */
+const getOwnerSchoolId = async (ownerType, ownerId) => {
+	const ownerModel = {
+		user: userModel,
+		course: courseModel,
+		teams: teamsModel,
+	};
+	const owner = await ownerModel[ownerType].findById(ownerId);
+	const schoolId = ownerType === 'teams' ? owner.schoolIds[0] : owner.schoolId;
```

Comment:
@CeEv  can you check this?

Reply:
Please use schoolId this is the school that create inital the team. The schoolIds are generic based on the added users. But this key is not the owner of the school. The files should generate for the owner of the team. But it is possible to switch the owner of the team over the team admin board. I am not sure how we can make it in sync with the s3.